src/wasm-interpreter.h

```diff
@@ -2422,23 +2422,24 @@ template<typename GlobalManager, typename SubType> class ModuleInstanceBase {
   protected:
     // Returns the instance that defines the memory used by this one.
     SubType* getMemoryInstance() {
-      if (instance.wasm.memory.imported()) {
-        return instance.linkedInstances.at(instance.wasm.memory.module).get();
-      } else {
-        return static_cast<SubType*>(&instance);
+      SubType* inst = static_cast<SubType*>(&instance);
```

Comment:
Why is this static cast needed?

Reply:
`instance` is of type `wasm::ModuleInstanceBase<GlobalManager, SubType>*`, while `linkedInstances` have `SubType*` so it needs casting. But I didn't need to cast directly, as we have a `wasm::ModuleInstanceBase<GlobalManager, SubType>::self()` method that returns does it for us. So I changed this part to use `instance.self()` instead.